lib/src/shared/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -390,6 +391,60 @@ public void stop() {
     @NonNull
     public Status getStatus() { return status.copy(); }
 
+    @NonNull
+    public Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
+        if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
+            throw new CouchbaseLiteException(
+                "Pending Document IDs are not supported on pull-only replicators.",
+                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Code.UNSUPPORTED);
+        }
+
+        final C4Replicator repl;
+        synchronized (lock) { repl = c4repl; }
+
+        if (repl == null) {
+            Log.i(DOMAIN, "%s: Call to getPendingDocumentIds with unstarted replicator", this);
+            return Collections.emptySet();
+        }
+
+        final Set<String> pending;
+        try { pending = c4repl.getPendingDocIDs(); }
```

Comment:
I think we may need to lock this as well as c4repl could be freed anytime?

Reply:
I think that is ok.  c4repl cannot be null, at this point, so the call itself is safe.  `C4Replicator` is responsible for making sure that its native companion object exists for the call.

Reply:
c4repl will not be null but the native C4Replicator will be freed (I haven't checked the code to see what could be happening).

Reply:
Right.  I tried, a while ago, in response to several bugs in that area, to fix that.  I *think* it is safer and clearer, now:
https://github.com/couchbase/couchbase-lite-java/blob/65969765ad12ded165d7468953612d5dbba86626/lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Replicator.java#L255